### DISCUSSION TITLE
Building universal binary on macOS

### DISCUSSION THREAD

**Discussion by pronebird** [tauri-apps/tauri#9419] [Q&A] [answered]
Hi,

I know that in the past (v1) tauri could build universal binaries on macOS using `tauri build --target universal-apple-darwin`. 

However it looks like that option was removed in v2, and so that begs the question, what's the alternative these days? Should I compile two `.app`s and then `lipo` two `My app.app/Contents/MacOS/<binary name>` manually?

```
 npm run tauri build --target universal-apple-darwin

> tauri
> tauri build universal-apple-darwin

    Running beforeBuildCommand `npm run build`

> build
> tsc && vite build

vite v5.2.7 building for production...
✓ 42 modules transformed.
dist/index.html                   0.37 kB │ gzip:  0.25 kB
dist/assets/index-CGgmDsQz.css    1.56 kB │ gzip:  0.67 kB
dist/assets/index-Bjmk0Prw.js   170.45 kB │ gzip: 55.64 kB
✓ built in 869ms
error: unexpected argument 'universal-apple-darwin' found
```

Also `tauri build --help` still states that it supports universal builds on macOS:

```
  -t, --target <TARGET>
          Target triple to build against.

          It must be one of the values outputted by `$rustc --print target-list` or `universal-apple-darwin` for a universal macOS application.
```

Best,
Andrej

**Comment by pronebird** [comment, marked as answer]
Actually I realized that npm obscures how parameters are passed into `tauri-cli` so the right way to pass flags is to use `--` delimiter:

```
npm run tauri build -- --target universal-apple-darwin
```

**Comment by PJUllrich** [comment]
Just an FYI that before running the build step, you also have add the `aarch64-apple-darwin` Rust target:

```bash
rustup target add aarch64-apple-darwin
```

If you only add the `x86_64-apple-darwin` target, for example as described in the [GitHub Action workflow example](https://v2.tauri.app/distribute/pipelines/github/#example-workflow) you will receive a build error.

So, the full GitHub Action step for building a universal apple build looks like this:

```yml
      - name: Install Rust targets for mac and build Tauri binaries
        if: matrix.os == 'macos-15'
        run: |
          rustup target add aarch64-apple-darwin
          rustup target add x86_64-apple-darwin
          bun install
          bun run tauri build --verbose --target universal-apple-darwin
```

**Reply by FabianLars** [reply]
Yeah it depends on your host architecture. Since the GitHub action runs on arm we only mention the x86_64 target there.
